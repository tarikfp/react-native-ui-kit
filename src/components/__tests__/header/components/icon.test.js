import * as React from "react";
import renderer from "react-test-renderer";
import HeaderIcon from "../../../header/components/icon";

describe("tests header icon component", () => {
  it("renders header icon with name and type", () => {
    const tree = renderer
      .create(<HeaderIcon name="arrow-left" type="MaterialCommunityIcons" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders header icon with icon with size", () => {
    const tree = renderer
      .create(
        <HeaderIcon name="arrow-left" type="MaterialCommunityIcons" size={45} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders header icon with color", () => {
    const tree = renderer
      .create(
        <HeaderIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          color="orange"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders header icon with alignment", () => {
    const tree = renderer
      .create(
        <HeaderIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          alignment="end"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders header icon with icon custom style", () => {
    const tree = renderer
      .create(
        <HeaderIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          style={{ marginLeft: 8 }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders header icon with custom wrapper style ", () => {
    const tree = renderer
      .create(
        <HeaderIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          wrapperStyle={{
            backgroundColor: "royalblue",
            width: 600,
            height: 50
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
