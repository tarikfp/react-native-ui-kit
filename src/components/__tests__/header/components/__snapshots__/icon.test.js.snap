// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests header icon component renders header icon with alignment 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests header icon component renders header icon with color 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "orange",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests header icon component renders header icon with custom wrapper style  1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "royalblue",
        "height": 50,
        "width": 600,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests header icon component renders header icon with icon custom style 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        Object {
          "marginLeft": 8,
        },
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests header icon component renders header icon with icon with size 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 45,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests header icon component renders header icon with name and type 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;
